### ISSUE TITLE
Where is the parameter \gamma

### ISSUE THREAD

**Issue by happynear** [s9xie/DSN#2]
In formulation (3), there is a factor \gamma. This parameter is setted to prevent the hinge loss to be 0. However, I haven't find this parameter in the code. 
The 0 loss is quite common in deep learning and this phenomenon is usually called "overfit". In deep learning, people usually use dropout to prevent the loss from getting to zero too early.

**Comment by sh0416** [comment]
@s9xie I am working on implementing your method. So, you mean that you don't explicitly use `gamma`, right? Actually, I am also curious about the other hyperparameter `alpha`, which requires exponential search space when layer increases. When I see your paper, you use relatively small architecture, i.e. 3-layer NN. How to tune this hyperparameter?

Thanks,
